b9lab-content/4-running-a-chain/04-node-api-and-cli.md

```diff
@@ -1 +1,464 @@
+  denom: stake
+pagination:
+  next_key: null
+  total: "0"
+```
+
+Yep, that's 10 stake.
+
+## CLI routing
+
+Now for a bit of Go code. How does this `simd` interact via the command-line interface? If you inspect the [`cosmos-sdk/simapp/simd/main.go`](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/simapp/simd/main.go) file:
+
+```go
+package main
+
+import (
+  "os"
+
+  "github.com/cosmos/cosmos-sdk/server"
+  svrcmd "github.com/cosmos/cosmos-sdk/server/cmd"
+  "github.com/cosmos/cosmos-sdk/simapp"
+  "github.com/cosmos/cosmos-sdk/simapp/simd/cmd"
+)
+
+func main() {
+  rootCmd, _ := cmd.NewRootCmd()
+
+  if err := svrcmd.Execute(rootCmd, simapp.DefaultNodeHome); err != nil {
+    switch e := err.(type) {
+    case server.ErrorCode:
+      os.Exit(e.Code)
+
+    default:
+      os.Exit(1)
+    }
+  }
+}
+```
+
+This [`cmd.NewRootCmd()`](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/simapp/simd/main.go#L13) function is the CLI handler and is imported via the [`"github.com/cosmos/cosmos-sdk/simapp/simd/cmd"`](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/simapp/simd/main.go#L9) line. In fact it can be found in the [`cosmos-sdk/simapp/simd/cmd/root.go`](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/simapp/simd/cmd/root.go#L39) file:
+
+```go
+func NewRootCmd() (*cobra.Command, params.EncodingConfig)
+```
+
+In it, [basic properties](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/simapp/simd/cmd/root.go#L51-L53) such as the application name are defined:
+
+```go
+rootCmd := &cobra.Command{
+    Use:   "simd",
+    Short: "simulation app",
+```
+
+In addition, observe that cobra is imported and [used for the CLI](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/simapp/simd/cmd/root.go#L144-L155) to redirect:
+
+```go
+rootCmd.AddCommand(
+    genutilcli.InitCmd(simapp.ModuleBasics, simapp.DefaultNodeHome),
+    genutilcli.CollectGenTxsCmd(banktypes.GenesisBalancesIterator{}, simapp.DefaultNodeHome),
+    genutilcli.MigrateGenesisCmd(),
+    genutilcli.GenTxCmd(simapp.ModuleBasics, encodingConfig.TxConfig, banktypes.GenesisBalancesIterator{}, simapp.DefaultNodeHome),
+    genutilcli.ValidateGenesisCmd(simapp.ModuleBasics),
+    AddGenesisAccountCmd(simapp.DefaultNodeHome),
+    tmcli.NewCompletionCmd(rootCmd, true),
+    NewTestnetCmd(simapp.ModuleBasics, banktypes.GenesisBalancesIterator{}),
+    debug.Cmd(),
+    config.Cmd(),
+)
+```
+
+Also explain [`simapp/app.go`](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/simapp/app.go) in which each module and keykeeper will be imported. The first thing you will see is a considerable [list of modules](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/simapp/app.go#L33-L44) that are used by most cosmos-sdk apps:
```

Comment:
The first sentence is a bit unclear. Do you mean to say, "Also, please look at ... " ?
```suggestion
Also, have a look at [`simapp/app.go`](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/simapp/app.go) in which each module and keykeeper will be imported. The first thing you will see is a considerable [list of modules](https://github.com/cosmos/cosmos-sdk/blob/d83a3bf92c9a84bddf3f5eb6692a1101c18b42f1/simapp/app.go#L33-L44) that are used by most cosmos-sdk apps:
```